src/js/models/geocoder/GoogleMapsAutocompleter.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+define(
+  ['backbone', 'gmaps', 'models/geocoder/Prediction'],
+  (Backbone, gmaps, Prediction) => {
+    /**
+    * @class GoogleMapsAutocompleter
+    * @classdes Integrate with the Google Maps Places Autocomplete API using the
+    * Google Maps AutocompleteService JS library.
+    */
+    return class GoogleMapsAutocompleter {
+      /**
+       * Google Maps service for interacting with the Places Autocomplete API.
+       */
+      autocompleter = new gmaps.places.AutocompleteService();
+
+      /**
+       * Use the Google Maps Places API to get place predictions based off of a
+       * user input string as the user types.
+       * @param {string} input - User input to search for Google Maps places.
+       * @returns {Prediction[]} An array of places that could be the result the
+       * user is looking for. Most often this comes in five or less results.
+       */
+      async autocomplete(input) {
+        try {
+          const response = await this.autocompleter.getPlacePredictions({
+            input,
+          });
+          return this.getPredictionsFromResults(response.predictions);
+        } catch (e) {
+          return [];
```

Comment:
Other than logging to the console (which I'd prefer not to do in case it gets logged in production), is there a place to log errors? Otherwise I'm unsure of the benefit there. I'd prefer to not show the user anything if its not an error that they can recover from. As far as I can tell from looking at the [Google Maps docs](https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompleteResponse), this will just return an empty response if the input is bad. So the only failure here would be a network failure